clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -94,7 +94,7 @@ private Utils() {}
 
     // This matches URIs of formats: host:port and protocol://host:port
     // IPv6 is supported with [ip] pattern
-    private static final Pattern HOST_PORT_PATTERN = Pattern.compile("^(?:[a-zA-Z][a-zA-Z\\d+-.]*://)?\\[?([0-9a-zA-Z\\-._%:]+)\\]?:([0-9]+)$");
+    private static final Pattern HOST_PORT_PATTERN = Pattern.compile("^(?:[a-zA-Z0-9+_.%-]*://)?\\[?([0-9a-zA-Z\\-._%:]+)]?:([0-9]+)$");
```

Comment:
I did mean `0-9a-zA-Z\\-%._` because that's what we had before. That's also what you have for the hostname part of the pattern, and I suggest the protocol part should use the same form and the same order of matching characters. Because it's not identical, people wanting to understand the change have a bit more work to do in order to check for equivalence.

Reply:
I see, I think I miss understand before. 🤔